File: rest/src/main/java/io/sparkled/viewmodel/exception/ViewModelConversionException.java
package io.sparkled.viewmodel.exception;

/**
 * Thrown when a view model conversion failure occurs. The message contained in the exception is
 * guaranteed to be user friendly.
 */
public class ViewModelConversionException extends RuntimeException {

    public ViewModelConversionException(String message) {
        super(message);
    }

    public ViewModelConversionException(String message, Throwable cause) {
        super(message, cause);
    }
}

File: rest/src/main/kotlin/io/sparkled/viewmodel/exception/ViewModelConversionException.kt
package io.sparkled.viewmodel.exception

/**
 * Thrown when a view model conversion failure occurs. The message contained in the exception is
 * guaranteed to be user friendly.
 */
class ViewModelConversionException : RuntimeException {

    constructor(message: String) : super(message) {}

    constructor(message: String, cause: Throwable) : super(message, cause) {}
}
